application/src/main/java/com/musicforall/services/recommendation/RecommendationService.java

```diff
@@ -40,10 +40,22 @@
         return trackService.getAllById(popularTracksIds);
     }
 
+    public Collection<Track> getRecommendedOrPopularTracks() {
```

Comment:
never use "OR" in method`s name. according to SRC method should do only one thing.